### PR TITLE
added authentication and routing

### PR DESCRIPTION
**Pull request by ShreyashMahajan**
In this PR, 
1. Authentication is added.
2. Routing is added.

### CODE REVIEW

File: src/pages/authentication/signUp/signUp.jsx [hunk 1]
```diff
@@ -11,42 +26,41 @@ export const SignUp = () => {
...
+                  <input type="number" name="contact" id="userContact" className="input-data" placeholder="contact number" onChange={(e) => inputChangeHandler(e)} />
                   <div className="gender-info">
                     <label htmlFor="gender">Gender</label>
                     <br />
-                    <input type="radio" name="gender" defaultValue="male" className="input-radio" />Male
-                    <input type="radio" name="gender" defaultValue="female" className="input-radio" />Female
-                    <input type="radio" name="gender" defaultValue="other" className="input-radio" />Other
+                    <input type="radio" name="gender" defaultValue="male" className="input-radio" onChange={(e) => inputChangeHandler(e)} />Male
+                    <input type="radio" name="gender" defaultValue="female" className="input-radio" onChange={(e) => inputChangeHandler(e)} />Female
+                    <input type="radio" name="gender" defaultValue="other" className="input-radio" onChange={(e) => inputChangeHandler(e)} />Other
```
Comment (Bhakti199) [inline src/pages/authentication/signUp/signUp.jsx:48]: you may use map over here.

**Review by Bhakti199 — APPROVED** [review]
- overall code looks fine to me.
- modular code.
- remove unwanted consoles.
- this pr is to merge.

File: src/components/productCard/ProductCard.jsx [hunk 2]
```diff
@@ -30,9 +30,9 @@ export const ProductCard = () => {
                                 </div>
                                 <p className="product--downpayment">Zero downpayment</p>
                                 <div className="product-cta">
-                                    <button className="product__btn" onClick={() => cartDispatch({ type: "ADD_TO_CART", payload: item })}>Buy
+                                    <button className="product__btn" onClick={() => addToCart(item)}>Buy
                                         Now</button>
-                                    <img src={wishListNotActiveImg} onClick={() => wishListDispatch({ type: 'ADD_WISHLIST', payload: item })} alt="" className="product-wishlist" />                                </div>
+                                    <img src={wishListNotActiveImg} onClick={() => addToWishlist(item)} alt="" className="product-wishlist" />                                </div>
```
Comment (Kothalkarpriya) [inline src/components/productCard/ProductCard.jsx:35]: Do use code formatting as it adds spacing where needed to look prettier

File: src/context/cartContext/cartContext.jsx [hunk 3]
```diff
@@ -1,16 +1,99 @@
...
 
-    const [cartState, cartDispatch] = useReducer(cartReducer, { cartList: [] });
+    const navigate = useNavigate();
+    const { isUserLoggedIn } = useAuth();
 
-    return <CartContext.Provider value={{ cartDispatch, cartState }}>
+    useEffect(() => {
+        if (status !== 'IDLE') return;
+        (async function () {
+            const { data, success, message } = await getCartDataService();
```
Comment (Kothalkarpriya) [inline src/context/cartContext/cartContext.jsx:22]: good you used async and await

**Review by Kothalkarpriya — APPROVED** [review]
Try to create small PR. You have added auth, cart and wishlist functionality in this one

File: src/pages/authentication/signUp/signUp.jsx (hunk 1 above)
Comment (ShreyashMahajan) [reply src/pages/authentication/signUp/signUp.jsx:48]: thank you, i will implement that

File: src/components/productCard/ProductCard.jsx (hunk 2 above)
Comment (ShreyashMahajan) [reply src/components/productCard/ProductCard.jsx:35]: ok

File: src/context/cartContext/cartContext.jsx (hunk 3 above)
Comment (ShreyashMahajan) [reply src/context/cartContext/cartContext.jsx:22]: thank you

**Pull request merged**